x-pack/solutions/observability/plugins/apm/public/components/app/settings/agent_configurations/agent_configuration_create_edit/settings_page/advanced_config_key_input.tsx

```diff
@@ -23,52 +31,84 @@ export function AdvancedConfigKeyInput({
   configKey: string;
   id: number;
   showLabel: boolean;
-  onChange: (newKey: string) => void;
-  checkIfAdvancedConfigKeyExists: (newKey: string, oldKey: string) => boolean;
+  revalidate: boolean;
+  onChange: ({ key, oldKey }: { key: string; oldKey: string }) => void;
+  checkIfAdvancedConfigKeyExists: (key: string) => boolean;
   checkIfPredefinedConfigKeyExists: (key: string) => boolean;
-  addValidationError: (key: string) => void;
+  addValidationError: (key: string, active: boolean) => void;
   removeValidationError: (key: string) => void;
 }) {
   // Handle key inputs with local state to avoid duplicated keys overwriting each other
   const [localKey, setLocalKey] = useState(configKey);
-  const [errorMsg, setErrorMsg] = useState<string | null>(null);
-  const isFormInvalid = localKey !== configKey && Boolean(errorMsg);
+  const [errorType, setErrorType] = useState<ErrorType>(ErrorType.None);
+  const [isFormInvalid, setIsFormInvalid] = useState(false);
 
   useEffect(() => {
     setLocalKey(configKey);
   }, [configKey]);
 
-  const getErrorMsg = (newKey: string) => {
-    if (newKey === '') {
-      return i18n.translate('xpack.apm.agentConfig.settingsPage.keyEmptyError', {
-        defaultMessage: 'Key cannot be empty',
-      });
-    }
-    if (checkIfPredefinedConfigKeyExists(newKey)) {
-      return i18n.translate('xpack.apm.agentConfig.settingsPage.keyPredefinedError', {
-        defaultMessage: 'This key is already predefined in the standard configuration above',
-      });
-    }
-    if (checkIfAdvancedConfigKeyExists(newKey, configKey)) {
-      return i18n.translate('xpack.apm.agentConfig.settingsPage.keyDuplicateError', {
-        defaultMessage: 'This key is already used in another advanced configuration',
-      });
+  const touched = useCallback((key: string) => key !== configKey, [configKey]);
```

Comment:
This doesn't need to be memoized with useCallback as it's not used by any component.